tripy/examples/diffusion/example.py

```diff
@@ -0,0 +1,285 @@
+    else:
+        diffusion_run_start = None
+        diffusion_run_end = None
+
+    # Upsample latent space to image with autoencoder
+    if args.verbose:
+        print(f"[I] Decoding latent...", end=" ")
+    vae_run_start = time.perf_counter() if args.verbose else None
+    x = vae_compiled(latent)
+    if args.verbose:
+        tp.default_stream().synchronize()
+        vae_run_end = time.perf_counter()
+        print(f"took {vae_run_end - vae_run_start} seconds.")
+    else:
+        vae_run_start = None
+        vae_run_end = None
+
+    # Evaluate output
+    run_end_time = time.perf_counter() if args.verbose else None
+    if args.verbose:
+        print(f"[I] Full script took {run_end_time - run_start_time} seconds.")
+
+    image_array = np.from_dlpack(tp.copy(x, tp.device("cpu"))).astype(np.uint8, copy=False)
+    image = Image.fromarray(image_array)
+
+    return image, [clip_run_start, clip_run_end, diffusion_run_start, diffusion_run_end, vae_run_start, vae_run_end]
+
+
+def print_summary(denoising_steps, times, verbose=False):
+    if not verbose or times is None or None in times:
+        return
+
+    stages_ms = [1000 * (times[i + 1] - times[i]) for i in range(0, 6, 2)]
+    total_ms = sum(stages_ms)
+    print("|-----------------|--------------|")
+    print("| {:^15} | {:^12} |".format("Module", "Latency"))
+    print("|-----------------|--------------|")
+    print("| {:^15} | {:>9.2f} ms |".format("CLIP", stages_ms[0]))
+    print("| {:^15} | {:>9.2f} ms |".format("UNet" + " x " + str(denoising_steps), stages_ms[1]))
+    print("| {:^15} | {:>9.2f} ms |".format("VAE-Dec", stages_ms[2]))
+    print("|-----------------|--------------|")
+    print("| {:^15} | {:>9.2f} ms |".format("Pipeline", total_ms))
+    print("|-----------------|--------------|")
+    print("Throughput: {:.2f} image/s".format(1000.0 / total_ms))
+
+
+def main():
+    default_prompt = "a beautiful photograph of Mt. Fuji during cherry blossom"
+    parser = argparse.ArgumentParser(
+        description="Run Stable Diffusion", formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("--steps", type=int, default=30, help="Number of denoising steps in diffusion")
+    parser.add_argument("--prompt", type=str, default=default_prompt, help="Phrase to render")
+    parser.add_argument("--out", type=str, default=None, help="Output filepath")
+    parser.add_argument("--fp32", action="store_true", help="Run the model in fp32 precision")
+    parser.add_argument("--seed", type=int, help="Set the random latent seed")
+    parser.add_argument("--guidance", type=float, default=7.5, help="Prompt strength")
+    parser.add_argument(
+        "--hf-token", type=str, default="", help="HuggingFace API access token for downloading model checkpoints"
+    )
+    parser.add_argument("--engine-dir", type=str, default="engines", help="Output directory for TensorRT engines")
```

Comment:
These would actually be Tripy executables rather than engines right?

Reply:
We should make the distinction since the former has extra metadata and so can't be directly deserialized by TRT.

Reply:
Oh I wasn't even aware of that. Yes, we should use "executable" globally ideally. I can make a separate PR for that.